File: U04/p3/collatz.py
def collatz(k):
	i = 0
	while k > 1:
		i += 1
		if k%2==0:
			k = k/2
		else:
			k = k*3
			k = k+1
	return i

for n in range(5,10):
	print collatz(n)

File: U04/p3/collatz.c
#include <stdio.h>

extern unsigned int collatz(unsigned int);

int main(int argc, char const *argv[])
{
	for (int i = 5; i < 10; ++i)
	{
		printf("%u \n",  collatz(i));
	}
	return 0;
}
